packages/ai-core/src/browser/window-blink-service.ts

```diff
@@ -0,0 +1,257 @@
+        this.blinkWithVisibilityAPI();
+    }
+
+    /**
+     * Blink the document title to get user attention.
+     */
+    private async blinkDocumentTitle(): Promise<void> {
+        const originalTitle = document.title;
+        const alertTitle = '🔔 Theia - Agent Completed';
+
+        let blinkCount = 0;
+        const maxBlinks = 6;
+
+        const blinkInterval = setInterval(() => {
+            if (blinkCount >= maxBlinks) {
+                clearInterval(blinkInterval);
+                document.title = originalTitle;
+                return;
+            }
+
+            document.title = blinkCount % 2 === 0 ? alertTitle : originalTitle;
+            blinkCount++;
+        }, 500);
+    }
+
+    /**
+     * Try to get user attention using the Page Visibility API.
+     */
+    private blinkWithVisibilityAPI(): void {
+        // If the page is hidden, try to trigger visibility change events
+        if (document.hidden) {
+            window.focus();
+        }
+
+        // Try to trigger a notification-like behavior
+        if ('Notification' in window && Notification.permission === 'granted') {
+            const notification = new Notification('Agent Completed', {
+                body: 'An AI agent has finished its task.',
+                icon: '/favicon.ico',
+                tag: 'agent-completion',
+                requireInteraction: false
+            });
+
+            setTimeout(() => {
+                notification.close();
+            }, 100);
+        }
+    }
+
+    /**
+     * Focus the window - basic fallback method.
+     */
+    private focusWindow(): void {
+        try {
+            window.focus();
+
+            // Try to scroll to top to create some visual movement
+            if (document.body.scrollTop > 0 || document.documentElement.scrollTop > 0) {
+                const currentScroll = document.documentElement.scrollTop || document.body.scrollTop;
+                window.scrollTo(0, 0);
+                setTimeout(() => {
+                    window.scrollTo(0, currentScroll);
+                }, 100);
+            }
+        } catch (error) {
+            console.debug('Could not focus window:', error);
+        }
+    }
+
+    /**
+     * Detect if running in Electron environment.
+     */
+    private detectElectronEnvironment(): boolean {
+        // Use Theia's environment detection first
+        if (environment.electron.is()) {
+            return true;
+        }
+
+        // Additional detection methods as fallback
+        return !!(
+            (window as unknown as { electronAPI?: unknown }).electronAPI ||
+            (window as unknown as { electron?: unknown }).electron ||
+            (window as unknown as { process?: { type?: string } }).process?.type ||
+            (typeof navigator !== 'undefined' && navigator.userAgent && navigator.userAgent.toLowerCase().includes('electron'))
+        );
+    }
```

Comment:
Same as in other class. If this is not sufficient i would at least at this to the core package and provide the functinality there and not calculate this in every class seperately.